src/tools/ScenarioMeasurement/Startup.Tests/StartupTests.cs

```diff
@@ -0,0 +1,98 @@
+using Microsoft.DotNet.PlatformAbstractions;
+using Reporting;
+using ScenarioMeasurement;
+using System;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.IO;
+using System.Resources;
+using Xunit;
+
+namespace Startup.Tests
+{
+    public class StartupTests
+    {
+        Logger logger = new Logger("test-startup.log");
+        string traceDirectory = Environment.CurrentDirectory;
+
+        [WindowsOnly]
+        public void TestWindowsTraceSession()
+        {
+            string sessionName = "test-windows-session";
+            string traceName = "test-windows-trace";
+            var session = new WindowsTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [LinuxOnly]
+        public void TestLinuxTraceSession()
+        {
+            string sessionName = "test-linux-session";
+            string traceName = "test-linux-trace";
+            var session = new LinuxTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [WindowsOnly]
+        public void TestProfileIteration()
+        {
+            string sessionName = "test-profile-iteration-session";
+            string traceName = "test-profile-iteration-trace";
+            var timeToMainParser = new TimeToMainParser();
+            var profileParser = new ProfileParser(timeToMainParser);
+            var profileSession = TraceSessionManager.CreateProfileSession(sessionName, traceName, traceDirectory, logger);
+            TestSession(profileSession, profileParser);
+        }
+
+        private void TestSession(ITraceSession session, IParser parser)
+        {
+            ProcessHelper.Result result;
+            string traceFilePath = "";
+            using (session)
+            {
+                session.EnableProviders(parser);
+                result = RunTestIteration();
```

Comment:
Do we even need to make a call to a sub-process here? We are not checking to see if that process shows up in the file, just to see if enabling providers works and we get something. Why not just put a sleep? That seems like it will make the test more robust and still test the same thing.

Reply:
Totally reasonable. We're indirectly getting test coverage of the process starting goo, which we could move to being an explicit test instead if we wanted.

I doubt we'll ever write tests that look for specific events - that's what the perf tests will ultimately do. So just ensuring that files get produced is not a bad idea.

Reply:
Cool! I was thinking ```dotnet --info``` is a bit closer to the real scenario... but less reasonable if we are just testing the trace sessions. Checking individual events will probably be implemented in tests of the kernel parser wrapper that will be added to Startup soon.